packages/clerk-js/src/core/clerk.ts

```diff
@@ -948,7 +952,36 @@ export default class Clerk implements ClerkInterface {
     this.#componentControls?.updateProps(props);
   };
 
-  #loadInStandardBrowser = async (): Promise<void> => {
+  #redirectToSyncWithPrimary = () => {
+    const q = new URLSearchParams({
+      redirect_url: `${window.location.href}?synced=true`,
+    });
+    window.location.replace(new URL(`/v1/client/sync?${q.toString()}`, `https://${this.domain}`).toString());
+  };
+
+  #stripSyncedQueryParam = (q: URLSearchParams) => {
+    q.delete('synced');
+    const queryString = q.toString() ? `?${q.toString()}` : '';
+    window.history.replaceState({}, '', `${window.location.pathname}${queryString}`);
+  };
+
+  #shouldSyncWithMain = () => {
+    const q = new URLSearchParams(window.location.search);
+    if (q.get('synced') !== 'true') {
+      this.#redirectToSyncWithPrimary();
+      return true;
+    }
+    this.#stripSyncedQueryParam(q);
+    return false;
+  };
+
+  #loadInStandardBrowser = async (): Promise<boolean> => {
+    if (this.#options.isSatellite) {
+      if (this.#shouldSyncWithMain()) {
+        return false;
+      }
+    }
```

Comment:
:upside_down_face: This can be simplified further.

```js
if (this.shouldSyncWithPrimary()) {
  this.syncWithPrimary();
}
// do whatever else
  return
}

// shouldSyncWithPrimary()
return isSatellite && !syncTrueParamInTheQuery
```